assets/js/order-script.js

```diff
@@ -78,48 +81,93 @@ jQuery(document).ready(function($) {
 				}
 			}
 		});
-	});
+	} );
 
-	$( ".wcpdf-data-fields .wpo-wcpdf-regenerate-document" ).on("click", function() {
-		
-		if ( window.confirm( wpo_wcpdf_ajax.confirm_regenerate ) === false ) {
-			return; // having second thoughts
-		}
+	// regenerate document
+	$( '#wpo_wcpdf-data-input-box' ).on( 'click', '.wcpdf-data-fields .wpo-wcpdf-regenerate-document', save_document_data );
 
-		$(this).addClass('wcpdf-regenerate-spin');
-		$form = $(this).closest('.wcpdf-data-fields');
+	// save document
+	$( '#wpo_wcpdf-data-input-box' ).on( 'click', '.wcpdf-data-fields .wpo-wcpdf-save-document', save_document_data );
 
-		// create an object with the data attributes
-		form_data_attributes = $form.data();
+	function save_document_data( e ) {
+		e.preventDefault();
 
-		// create a serialized string with the form inputs name/value
-		serialized = $form.find(":input:visible:not(:disabled)").serialize();
+		$form       = $(this).closest('.wcpdf-data-fields');
+		let action     = $(this).data('action');
+		let nonce      = $(this).data('nonce');
+		let data       = $form.data();
+		let serialized = $form.find(":input:visible:not(:disabled)").serialize();
 
 		// Make sure all feedback icons are hidden before each call
 		$form.find('.document-action-success, .document-action-failed').hide();
 
-		$.ajax({
+		// block UI
+		if( action == 'save' ) {
+			$form.block( {
+				message: null,
+				overlayCSS: {
+					background: '#fff',
+					opacity: 0.6
+				}
+			} );
+		} else if( action == 'regenerate' ) {
+			if ( window.confirm( wpo_wcpdf_ajax.confirm_regenerate ) === false ) {
+				return; // having second thoughts
+			}
+	
+			$form.find('.wpo-wcpdf-regenerate-document').addClass('wcpdf-regenerate-spin');
+		}
+
+		$.ajax( {
 			url:                wpo_wcpdf_ajax.ajaxurl,
 			data: {
-				action:         'wpo_wcpdf_regenerate_document',
-				security:       $(this).data('nonce'),
+				action:         'wpo_wcpdf_'+action+'_document',
+				security:       nonce,
 				form_data:      serialized,
-				order_id:       form_data_attributes.order_id,
-				document_type:  form_data_attributes.document,
+				order_id:       data.order_id,
+				document_type:  data.document,
+				action_type:    action,
 			},
 			type:               'POST',
 			context:            $form,
 			success: function( response ) {
 				if ( response.success ) {
-					$(this).find('.document-action-success').show();
+					if( action == 'save' ) {
+						$form.find(".read-only").show();
+						$form.find(".editable").hide();
+						$form.find(':input').attr('disabled', true);
+						$form.find('.wpo-wcpdf-document-buttons').hide();
+
+						// update document DOM data
+						let wrapper = data.document + '-fields .wrapper';
+						$form.load( document.URL + ' .' + wrapper, function() {
```

Comment:
That's quite a big AJAX request and I'm not sure if this will always work as expected. It's quite simple though (which I really like), so maybe we should just try this out. If it causes any issues with specific setups we can always pass the HTML with the success response directly (which requires a bit more code separation server side but is much lighter in usage).

Reply:
Sounds good to me :+1: